experiment/dispatcher.py

```diff
@@ -156,7 +160,8 @@ def dispatcher_main():
     trials = build_images_for_trials(experiment.fuzzers, experiment.benchmarks,
                                      experiment.num_trials,
                                      experiment.preemptible,
-                                     experiment.config['concurrent_builds'])
+                                     experiment.config['concurrent_builds'],
+                                     experiment.config['worker_pool_name'])
```

Comment:
1. Won't this line and the line above fail if these two keys do not exist?
2. I reckon `concurrent_builds` was not a mandatory field in the configuration YAML, why won't the old code break?

Reply:
Also, the current implementation passes `concurrent_builds` and `worker_pool_name` to all functions along the way to code that uses them, which makes those functions bloated.
Alternatively, I can prepare them in [experiment/run_experiment.py](https://github.com/google/fuzzbench/pull/1535/files#diff-f29fa4fe4e55f233de1357ff219fd6df4fc408fcbdd0406f3fb5a022e11c670b) and  define them as ENV variables in [`dispatcher-startup-script-template.sh`](https://github.com/google/fuzzbench/blob/master/experiment/resources/dispatcher-startup-script-template.sh).
However, that will globally expose those two variables.
3. Which one would you prefer?

Reply:
4. I thought the old code already ensures we can config `concurrent_builds`, as it parses `concurrent_builds` from YAML, and when it is `None`, [`builder.py` uses the default value](https://github.com/google/fuzzbench/pull/1535/files#diff-30d6caa95ef383e7b9176f3f0d64ee98f6dd5b1c466ab03b44492509763661beL114).

Reply:
> I think you introduced it right? 

Nah, the credit belongs to [8692e](https://github.com/google/fuzzbench/commit/8692edd60879f64fc010199676a6cd7cba5f7180) and [66980](https://github.com/google/fuzzbench/commit/669805b3b107c1b13c27c636a3e964ad929df0ba).

I was curious why it has not broken because 
1. This `experiment.config` is simply [a dictionary constructed with the content of `experiment-config.yaml`](https://github.com/google/fuzzbench/blob/5624dfea608c1ff55a11f82ac8f724877b7c85fa/experiment/dispatcher.py#L96). It does not set the default value.
2. It uses a direct key query (`experiment.config['concurrent_builds']`) instead of the `get` method (`experiment.get('concurrent_builds')`). If that key does not exist, the key query will break instead of returning `None`.
3. Many of our `experiment-config.yaml` does not have `concurrent_builds`, for example, [this one](https://github.com/google/fuzzbench/blob/5624dfea608c1ff55a11f82ac8f724877b7c85fa/service/experiment-config.yaml).
So I was curious why it has not broken so far.